lib/src/logical_replication_messages.dart

```diff
@@ -15,63 +15,110 @@ import 'types.dart';
 abstract class LogicalReplicationMessage
     implements ReplicationMessage, ServerMessage {}
 
-class XLogDataLogicalMessage extends XLogDataMessage {
-  late final LogicalReplicationMessage _message;
+class XLogDataLogicalMessage implements XLogDataMessage {
+  @override
+  // this and others are `late` to comply with super class
+  late final Uint8List bytes;
 
   @override
-  LogicalReplicationMessage get data => _message;
+  late final DateTime time;
 
-  XLogDataLogicalMessage(Uint8List bytes) : super(bytes) {
-    _message = parseLogicalReplicationMessage(this.bytes);
-  }
+  @override
+  late final LSN walEnd;
+
+  @override
+  late final LSN walStart;
+
+  @override
+  final LSN walDataLength;
+
+  late final LogicalReplicationMessage message;
+
+  @override
+  LogicalReplicationMessage get data => message;
+
+  XLogDataLogicalMessage({
+    required this.message,
+    required this.bytes,
+    required this.time,
+    required this.walEnd,
+    required this.walStart,
+    required this.walDataLength,
+  });
 
   @override
   String toString() => super.toString();
 }
 
-LogicalReplicationMessage parseLogicalReplicationMessage(Uint8List bytesList) {
+/// Tries to check if the [XLogDataMessage.bytes] is a [LogicalReplicationMessage]
+/// If so, it'll return
+XLogDataMessage tryParseLogicalReplicationMessage(XLogDataMessage message) {
```

Comment:
In my experience `tryX` usually returns `T?`: either `T` if successful, or `null` if not. This method seems to be better places on the `XLogDataMessage` (either directly or through an extension method) and named `cast` or `castToLogicalReplication`, especially if those message have a `LogicalReplicationMessage` interface or superclass. wdyt?

Reply:
Good catch and I totally agree with you.  

I added `XLogDataMessage.parse` and made `tryParseLogicalReplicationMessage` returns `LogicalReplicationMessage?`.  Let me know if it looks better.

Reply:
looks good, thank you!